### PR TITLE
Add types npm-registry-fetch, libnpmsearch, and pacote

### PR DESCRIPTION
**Pull request by joelspadin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/pacote/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+}
+
+export interface PackageVersion {
+    name: string;
+    version: string;
+    dependencies?: Record<string, string>;
+    optionalDependencies?: Record<string, string>;
+    devDependencies?: Record<string, string>;
+    peerDependencies?: Record<string, string>;
+    directories: {};
```
Comment (ChaosinaCan) [inline types/pacote/index.d.ts]: [npm's documentation for this field](https://github.com/npm/registry/blob/master/docs/REGISTRY-API.md#version) literally says "???". 🤷‍♀️

In every request I've tried, it's always shown up as an empty object.

File: types/pacote/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+// Project: https://github.com/npm/pacote#readme
+// Definitions by: Joel Spadin <https://github.com/ChaosinaCan>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+/// <reference types="node" />
+
+import npmFetch = require('npm-registry-fetch');
+
+export import extract = require('./extract');
```
Comment (uniqueiniquity) [inline types/pacote/index.d.ts]: For readability, I'd suggest doing:
```ts
import extract = require('./extract');
import manifest = require('./mainfest');
...

export { extract, manifest, ... };
```

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/pacote/index.d.ts (hunk 2 above)
Comment (ChaosinaCan) [reply types/pacote/index.d.ts]: Good idea. Updated.

**Review by uniqueiniquity — APPROVED** [review]

**Comment by ChaosinaCan** [comment]
Thanks!

**Pull request merged**